### ISSUE TITLE
Accuracy drop when merging Conv with BN

### ISSUE THREAD

**Issue by Shishuii** [Xilinx/brevitas#388]
Hi, I have a model with testing accuracy of 93.88 %. But when I merge the Conv with BN , my accuracy becomes 88.91. 
Below is my merging function.
````
def merge_conv_bn_model(model):
    conv_block_layers = [layer for layer in model.modules() if isinstance(layer,models.mobilenet_brevitas.ConvBlock)]
    model.eval()
    model.cpu()
    for layer in conv_block_layers:       
        merge_bn(layer.conv, layer.bn, output_channel_dim=0)  # function brevitas.nn.utils.merge_bn
    model_fused.load_state_dict({k: v for k, v in model.state_dict().items() if 'bn' not in k})#skippinh bn params
    print('layer fused ')
````
It would be great if someone can help me out, or explain the reason for accuracy drop.

**Comment by volcacius** [comment]
Hi @Shishuii,

If your conv is quantized merging is not necessarily a lossless operation. This is especially true if the conv is set to per-tensor weight quantization (as it is by default) and not per-channel (weight_scaling_per_output_channel=True). How is the conv layer declared? On top of this the merge_bn function is not particularly smart in the way it's implemented, even for the cases where it can be lossless there will be differences.

**Comment by Shishuii** [comment]
Hi @volcacius ,

My neural network follows mobilenetV1 architecture. Some layers have per-tensor quantization while few have per-channel. I am a bit confused regarding writing a custom batchnorm fusion function. What will be the scales of folded weights? Also, it will be very helpful if you can suggest some ways to reduce the accuracy drop that I can try.

Thank you for your time.

**Comment by volcacius** [comment]
Something that I'm noticing now, it seems to me that you are:
- First merging the BN layers into conv `merge_bn(layer.conv, layer.bn, output_channel_dim=0)  # function brevitas.nn.utils.merge_bn`
- The loading the model state_dict and ignoring the BN parameters ` model_fused.load_state_dict({k: v for k, v in model.state_dict().items() if 'bn' not in k})#skippinh bn params`

What happens this way is that you are simply merging the default BN parameters and then throwing away the ones you learned during the training process. You should first load the state dict and then perform merging. Once you do so, you should also avoid going through the BN layers in your forward pass, to avoid computing them twice.

**Comment by Shishuii** [comment]
Hi @volcacius, As suggested by you I loaded the model before merging. 
And now the accuracy drop is pretty less ( 91.25 (before merging)->91.09(after merging) ).
I will further look into reducing the slight accuracy gap.

Thank you very much for your suggestion and time.
